### DISCUSSION TITLE
Send a variable (from a card) using the button (of a card) to get results from a table (using that variable) in another page

### DISCUSSION THREAD

**Discussion by iwanhoogendoorn** [Budibase/budibase#11097] [Help]
I have a table with the name STUDENTS.
These STUDENTS have a COMPUTER assigned with an IP ADDRESS.
This information is retrieved from the STUDENTS table and displayed on a card.

When I click on the button on the card I want to use the IP ADDRESS as a variable, to retrieve more information (inside another page) on the COMPUTER that is stored in another table (named COMPUTERS).

How do I achieve this goal in Budibase? Is this even possible with Budibase?

Any help is appreciated!

**Comment by melohagan** [comment]
Hey @nsxninja 

I imagine what you are trying to do is possible, but I'm not quite certain exactly what you want. 
Have you had a look through the docs at all?

You wouldn't mind providing some screenshots to highlight what you have, and what you would like to see?

**Comment by R2bEEaton** [comment]
I'm not sure if this is exactly what you mean, but maybe this related answer might help?
https://github.com/Budibase/budibase/discussions/11098#discussioncomment-6361652

If you're fine with exposing the IP addresses in a URL parameter, you could do the following:
1. On your card, make it clickable or add a button navigating to a new screen with a URL parameter for IP Address, maybe `/computers/:ip`. The link to navigate to should be `/computers/{{ New Card.xyz.IP_ADDRESS }}` but obviously change it to however your data is named.
2. On the computers screen, add a filter for your data source where `IP_ADDRESS` is equal to a binding with `{{ URL.ip }}`.

Let me know if this is what you wanted or if you need more help setting it up!

**Reply by iwanhoogendoorn** [reply]
Thanks for your answer, but I think I will need a bit more detailed instructions :-)

**Reply by R2bEEaton** [reply]
No problem. Did you take a look at my other answer? I went into more detail there. What specifically do you need help with?

**Reply by iwanhoogendoorn** [reply]
I have seen the query question, and that one is clear, I am not using the Budibase DB, but I am using PostgreSQL.
I still have some issues with my queries. I just don't know how I need to use the URL IP in there.

I tried it but it did not work ....

**Reply by R2bEEaton** [reply]
Ok. So what's your structure? Are you using a PostgreSQL custom query or are you querying a regular table with a table?

### If it's a regular table and you don't need something more complicated

On the data source component or table block in the Budibase builder, there should be a filter button. Press that and set the filtering to something like this.

![image](https://github.com/Budibase/budibase/assets/34921506/86b0fc60-db20-49b3-b9e6-ae6618d5a940)

### If you need to add the IP address to the actual custom query itself

You can add a query binding like below. This is for MS SQL but I'd imagine it would be similar for PostgreSQL.

![image](https://github.com/Budibase/budibase/assets/34921506/0114a9f8-5169-48fc-bfe8-a4a3b0a206c1)

Then choose your custom query in your data provider in the Budibase builder and click the gear icon to set the bindings. Set IP to `{{ URL.ip }}` as explained before. Then wherever you have the `{{ IP }}` binding in your custom query will be replaced with the IP from the URL.

If you need more help, please provide some more info about what you're trying to do and how your providers are working. Thanks!
